src/python/pants/option/custom_types.py

```diff
@@ -306,3 +307,8 @@ def create(cls, value):
 
   def __repr__(self):
     return '{} {}'.format(self.action, self.val)
+
+
+class Conjunction(enum('conjunction', ['or', 'and'])):
```

Comment:
While having a `Conjuction` enum is elegant, I'm a bit concerned that it obfuscates what it is being used for in the context of PathGlobs expansion.

I think that it would probably be much clearer for this enum to be something like "all_match" or "any_match", even if that meant that it was a single-use type. That would make it significantly more self-documenting.

Reply:
Done in b5eec2d48732c77f7def517e49e6046a6334ad8c! Totally agreed that this is much clearer, I definitely *don't* want to introduce "algebra" we don't need yet -- my intention with the class was for this sole purpose so I'm 100% behind the rename.